app/models/my_academics/semesters.rb

```diff
@@ -153,22 +153,41 @@ def merge_semesters_count(data)
     end
 
     def map_enrollments(enrollment_term, term_id)
-      enrollment_term.map do |course|
-        next unless course[:role] == 'Student'
-        next if law_student? && !academic_careers.include?(course[:academicCareer])
-        mapped_course = course_info course
+      enrollment_term.map do |enrollment|
```

Comment:
Rather than Array#map + Array#compact, consider Array#filter which seems to be what you're actually trying to do

Reply:
@developish - Yeah i could make it filter separate of processing. Even break it up into two methods.